detox/src/devices/AttachedAndroidDriver.js

```diff
@@ -0,0 +1,38 @@
+const _ = require('lodash');
+const AAPT = require('./android/AAPT');
+const Emulator = require('./android/Emulator');
+const AndroidDriver = require('./AndroidDriver');
+
+class AttachedAndroidDriver extends AndroidDriver {
+
+  constructor(client) {
+    super(client);
+
+    this.aapt = new AAPT();
+    this.emulator = new Emulator();
+  }
+
+  async acquireFreeDevice(name) {
+    const adbDevices = await this.adb.devices();
```

Comment:
This is an almost exact copy of the logic in EmulatorDriver (and we hate duplicate code of course 😬 ) . Why not merge in into a utility function inside `AndroidDriver` ?